### ISSUE TITLE
Light simulation issue 1: Noise

### ISSUE THREAD

**Issue by YifanC** [DUNE/larnd-sim#196]
The simulation generates empty waveform for padding in order to produce light readout for all channels. Previously the padding waveforms look like this, because noise was not added.
![empty_light_wvfm_no_noise](https://github.com/DUNE/larnd-sim/assets/25907864/2e1e8b2a-8b4a-4ddf-9569-0405f93ddc84)

Adding noise in [such way](https://github.com/DUNE/larnd-sim/blob/49299af6a4c67faa83fcbfc160f2bb77b82d54ca/cli/simulate_pixels.py#L725) produces empty waveforms with noise. However, you can notice the noise in [the pre-trigger window](https://github.com/DUNE/larnd-sim/blob/49299af6a4c67faa83fcbfc160f2bb77b82d54ca/larndsim/detector_properties/2x2_mod2mod_variation.yaml#L61) is different, although they are not always different by eye in a single channel.
![empty_light_wvfm_separate_noise_all](https://github.com/DUNE/larnd-sim/assets/25907864/4fde5743-417a-4805-95fc-65b67d4a21dc)
![empty_light_wvfm_separate_noise_ch3](https://github.com/DUNE/larnd-sim/assets/25907864/855ffe10-b7db-4bc6-8c3c-3e09d89a38e4)

The way how light simulation in develop ([71a493c](https://github.com/DUNE/larnd-sim/commit/71a493cad33a13ec731b7c40b3df06e67a47fbda)) handles the signal and the noise is to join [the pre-trigger](https://github.com/DUNE/larnd-sim/blob/57d1e09b211ea78760f4d88359540399a4621a2a/larndsim/light_sim.py#L587), [the "signal"](https://github.com/DUNE/larnd-sim/blob/57d1e09b211ea78760f4d88359540399a4621a2a/cli/simulate_pixels.py#L1022) and [the post-signal](https://github.com/DUNE/larnd-sim/blob/57d1e09b211ea78760f4d88359540399a4621a2a/larndsim/light_sim.py#L597) and their noises (disjoint randomness). Especially the "signal" for thresholding and beam trigger are different. The pre-trigger part should always exist in the current scope, but the post-signal part may not be. 

Here, I prepose [to add noise](https://github.com/DUNE/larnd-sim/blob/49299af6a4c67faa83fcbfc160f2bb77b82d54ca/larndsim/light_sim.py#L606) only after padded signal is in shape. I think this should be an improvement in general.
![empty_light_wvfm_fixed_noise_all](https://github.com/DUNE/larnd-sim/assets/25907864/69ece1a9-db8c-4404-bb01-3298ed08c483)

Tag @mjkramer @krwood @liviocali @AWh1t3 for comments.

**Comment by AWh1t3** [comment]
I agree that generating noise three separate times for one waveform is unnecessarily complicated, and it would make perfect sense to just add the noise overlay to the completed waveform once at the end!

**Comment by AWh1t3** [comment]
The noise generating function is a reversed fast fourier transform. The noise files I've created are intended to be applied to a waveform with a very specific length and sample frequency. I think (I need to run a few trials to be sure) that the noise issues we've seen in terms of amplitude, at least, come from applying this function to a shorter sample, and potential issues with the code intended to allow that action. Thus, just generating noise for the final, fully-realized waveform would completely remove the potential for errors of this type.
